drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java

```diff
@@ -146,22 +137,17 @@ private Path writeToFile( final Path resource,
                               final String comment,
                               boolean create ) {
         final SessionInfo sessionInfo = getSessionInfo( sessionId );
-        String userAction = "UPDATING";
+
+        log.info("USER: {} {} asset [{}]", sessionInfo.getIdentity().getIdentifier(), create ? "CREATING" : "UPDATING", resource.getFileName());
+
+        final org.uberfire.java.nio.file.Path nioPath = Paths.convert( resource );
         if ( create ) {
-            userAction = "CREATING";
+            ioService.createFile( nioPath );
         }
-        log.info( "USER:" + sessionInfo.getIdentity().getIdentifier() + " " + userAction + " asset [" + resource.getFileName() + "]" );
-
-        OutputStream outputStream = null;
-        try {
-            final org.uberfire.java.nio.file.Path nioPath = Paths.convert( resource );
-            if ( create ) {
-                ioService.createFile( nioPath );
-            }
```

Comment:
Should we move both:
- `Paths.convert(resource)`
- `ioService.createFile(nioPath)`

out of the try catch block? Especially the second one informs in method signature about quite a lot possible runtime exceptions.

What is your opinion? Was this change intentional? I know general rule is to keep try catch block scope as small as possible. However here we update existing code base what makes our situation difficult. Did you compared what happens when user tries to upload a xls scorecard that has an existing name before and after your PR please?

Reply:
@jomarko  The change was intentional, and the reason for that change is `ioService.newOutputStream()` methods requires `nioPath` as paramenter. 
Please consider `ExceptionUtilities.handleException( e );`, which is the action performed if an expection is catched.
If the expection is unchecked (instance of RuntimeExpection), it is simply re-thrown, and this is exactly the case of the two methods you mentioned.

Reply:
The difference I see is the logger message inside `ExceptionUtilities.handleException( e );` will be no more reached. 
So, I don't see problems to move 
       ` if ( create ) {
            ioService.createFile( nioPath );
        }`
Inside the `try-catch` for this resaon.
`nioPath` I don't see a way to move it in `try-catch`.

Reply:
ok, thank you @yesamer